src/controller.py

```diff
@@ -103,7 +106,8 @@ def open_file(self):
     def load_stl(self, filename, colorize=False):
         if filename is None or filename == "":
             filename = self.model.opened_stl
-        stl_actor = gui_utils.createStlActorInOrigin(filename, colorize)
+        s = sett()
+        stl_actor = gui_utils.createStlActorInOrigin(s.slicing.copy_stl_file, colorize)
```

Comment:
you have a parameter, use it.

Reply:
Specify what do you mean?

Reply:
you have filename here, why did you change on s.slicing.copy? How to open the original then?

Reply:
load_stl method should load stl that in the filename. The method should not know about any settings.

Reply:
If only the input filename parameter is used in this method, then the self.model.opened_stl parameter will be overwritten with the path to the model copy. The opened_stl parameter is used in other places where the original model is required. In order not to lose the link to the original model after colorize(), after each use of the load_stl() method, it will be necessary to write in self.model.opened_stl the path from s.slicing.stl_file or in places where self.model.opened_stl is used, replace it with s.slicing.stl_file. Will any of these options work or should I do something else?

Reply:
line 111. I see. Not so beautiful. I think it is better to refactor the logic. Check where do we use opened_stl. Might be better to add a button to load "default stl" before the slicing or return filename to the parent func and save it in model there.